src/hooks/useFilterPossiblyMaliciousPositions.ts

```diff
@@ -0,0 +1,62 @@
+import { Interface } from '@ethersproject/abi'
+import { Token } from '@uniswap/sdk-core'
+import ERC20ABI from 'abis/erc20.json'
+import { Erc20Interface } from 'abis/types/Erc20'
+import { useMultipleContractSingleData } from 'lib/hooks/multicall'
+import { useMemo } from 'react'
+import { PositionDetails } from 'types/position'
+import { hasURL } from 'utils/urlChecks'
+
+import { useDefaultActiveTokens } from './Tokens'
+
+const ERC20Interface = new Interface(ERC20ABI) as Erc20Interface
+
+export function useFilterPossiblyMaliciousPositions(positions: PositionDetails[]): PositionDetails[] {
+  const tokens = useDefaultActiveTokens()
+
+  const nonListPositionTokenAddresses = useMemo(
+    () =>
+      Array.from(
+        new Set(positions.reduce((acc, position) => acc.concat(position.token0, position.token1), [] as string[]))
```

Comment:
`Set` deduplicates address set (i.e., if you've got 50 positions with the `WETH` address

Reply:
can you make this a separate function? like `getUniqueAddressesFromPositions` or something?